monsterCatalog.py

```diff
@@ -98,18 +99,68 @@ def generate_picture_id_url():
 
 def s3_store_picture(picture_id, picture):
     """Store picture in S3."""
-    session = boto3.session.Session(
+    s3_session = boto3.session.Session(
         aws_access_key_id=app.config['AWS_ACCESS_KEY_ID'],
         aws_secret_access_key=app.config['AWS_SECRET_ACCESS_KEY']
     )
-    s3 = session.resource(service_name='s3')
+    s3 = s3_session.resource(service_name='s3')
 
     s3.Bucket(app.config['AWS_S3_BUCKET']).put_object(
         Key=picture_id,
         Body=picture,
         ContentType='image/png')
 
 
+@app.route('/edit/<int:monster_id>', methods=['GET', 'POST'])
+def edit_monster(monster_id):
+    monster = db_session.query(Monster).filter_by(id=monster_id).one()
+    current_user_id = flask.session.get('user_id')
+    if monster is None:
+        flask.redirect('/')
+    if current_user_id is None:
+        flask.redirect('/signin')
+    # monster = monster_query[0]
+    if current_user_id != monster.creator:
+         # Add error message here
+        flask.redirect('/')
+
+    if flask.request.method == 'GET':
+        return flask.render_template('create.html',
+                                     name=monster.name,
+                                     intentions=monster.intentions,
+                                     diet=monster.diet,
+                                     enjoys=monster.enjoys,
+                                     picture_url=monster.picture,
+                                     submit_button_text='SUBMIT EDIT')
+
+    elif flask.request.method == 'POST':
```

Comment:
You might put the stuff in this branch of the flow into a function like `handle_edit_monster` to keep the method readable.